src/java/org/apache/cassandra/service/reads/ReadCallback.java

```diff
@@ -159,7 +162,25 @@ public int blockFor()
     public void onResponse(Message<ReadResponse> message)
     {
         assertWaitingFor(message.from());
+        
         resolver.preprocess(message);
+        
+        // Track the response size from replica to coordinator (only for actual remote responses)
+        // We record metrics after preprocess() to ensure we only track valid, accepted responses
+        // Only track messages that:
+        // 1. Are from remote nodes (message.from() != null)
+        // 2. Have a valid payload
+        // 3. Have a proper serializer (excludes test messages)
+        // 4. Are not MultiRangeReadResponse or LocalSubrangeResponse (which throw UnsupportedOperationException)
+        if (message.from() != null 
+            && message.payload != null
+            && message.verb().serializer() != null
+            && !(message.payload instanceof MultiRangeReadResponse)
+            && !message.payload.getClass().getSimpleName().equals("LocalSubrangeResponse"))
```

Comment:
Making LocalSubrangeResponse `protected` and using `instanceof` would be better for type safety and performance.